File: server/Simulation/Net/RequestSpawnPacket.cs
using System;
using server.ECS;
using server.Enums;

namespace server.Simulation.Net;

public class RequestSpawnPacket
{
    public NTT Requester { get; set; }
    public NTT Target { get; set; }

    public static Memory<byte> Create(NTT requester, NTT target)
    {
        using var writer = new PacketWriter(PacketId.RequestSpawnPacket);
        writer.WriteNtt(requester)
              .WriteNtt(target);
        return writer.Finalize();
    }

    public static RequestSpawnPacket Read(Memory<byte> buffer)
    {
        var reader = new PacketReader(buffer);
        var header = reader.ReadHeader(); // Skip header

        return new RequestSpawnPacket
        {
            Requester = reader.ReadNtt(),
            Target = reader.ReadNtt()
        };
    }
}

File: pixiejsClient/src/app/network/packets/RequestSpawnPacket.ts
import { EvPacketWriter } from '../EvPacketWriter';
import { EvPacketReader } from '../EvPacketReader';
import { PacketHeader } from '../PacketHeader';
import { PacketId } from '../PacketHandler';

export class RequestSpawnPacket {
    header: PacketHeader
    requester: string
    target: string

    constructor(header: PacketHeader, requester: string, target: string) {
        this.header = header
        this.requester = requester
        this.target = target
    }

    static create(requesterId: string, targetId: string): ArrayBuffer {
        const writer = new EvPacketWriter(PacketId.RequestSpawnPacket);
        writer
            .Guid(requesterId) // Requester NTT
            .Guid(targetId)    // Target NTT
            .FinishPacket();
        return writer.ToArray();
    }

    static fromBuffer(buffer: ArrayBuffer): RequestSpawnPacket {
        const reader = new EvPacketReader(buffer);
        const header = reader.Header();
        const requester = reader.Guid();
        const target = reader.Guid();
        return new RequestSpawnPacket(header, requester, target);
    }
}